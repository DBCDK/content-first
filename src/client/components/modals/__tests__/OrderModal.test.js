/* eslint-disable no-undefined */
import React from 'react';
import {
  OrderModal,
  mapStateToProps,
  mapDispatchToProps
} from '../OrderModal.container';
import renderer from 'react-test-renderer';

jest.mock('../../general/Spinner.component', () => 'Spinner');
jest.mock('../Modal.component', () => 'Modal');
jest.mock('../../general/BookCover.component', () => 'BookCover');

const sampleBooks = [
  {pid: 'pid1', title: 'title1', creator: 'creator1'},
  {pid: 'pid2', title: 'title2', creator: 'creator2'},
  {pid: 'pid3', title: 'title3', creator: 'creator3'}
];
const sampleBranches = [];
function checkOrderModal(books) {
  const tree = renderer
    .create(<OrderModal branches={sampleBranches} orders={books} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
  return tree;
}

describe('OrderModal', () => {
  describe('render function', () => {
    it('renders initial request', () => {
      checkOrderModal(sampleBooks);
    });
  });
  it('has a working mapStateToProps', () => {
    expect(
      mapStateToProps({
        orderReducer: {orders: {}, pickupBranches: []}
      })
    ).toEqual({branches: [], currentBranch: undefined, orders: []});
  });
  it('has a working mapDispatchToProps', () => {
    expect(Object.keys(mapDispatchToProps(() => {}))).toEqual([
      'onChangeBranch',
      'onStart',
      'onClose'
    ]);
  });
});
