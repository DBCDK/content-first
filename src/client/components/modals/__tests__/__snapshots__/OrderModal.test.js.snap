// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderModal render function renders initial request 1`] = `
<Modal
  className="add-to-list--modal"
  doneText="JA TAK, BESTIL NU"
  header="BESTIL"
  onClose={undefined}
  onDone={[Function]}
>
  <div>
    <div
      className="form-group"
    >
      <strong>
        Du er ved at bestille
         3 bøger
        :
      </strong>
      <div
        style={
          Object {
            "background": "white",
            "marginTop": 10,
            "maxHeight": 250,
            "overflowX": "hidden",
            "overflowY": "scroll",
            "scroll": "auto",
          }
        }
      >
        <div
          className="row short-list-page"
        >
          <div
            className="col-12"
            style={
              Object {
                "borderTop": "1px solid #ccc",
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          >
            <span
              style={
                Object {
                  "float": "left",
                  "height": 60,
                  "marginRight": 10,
                }
              }
            >
              <BookCover
                book={
                  Object {
                    "creator": "creator1",
                    "pid": "pid1",
                    "title": "title1",
                  }
                }
                style={
                  Object {
                    "width": "unset",
                  }
                }
              />
            </span>
            <div
              style={
                Object {
                  "display": "inline-block",
                  "float": "right",
                  "height": 60,
                  "marginRight": 10,
                }
              }
            >
              <small>
                <span
                  style={
                    Object {
                      "display": "inline-block",
                      "textAlign": "center",
                    }
                  }
                >
                  <Spinner
                    size={12}
                    style={
                      Object {
                        "marginTop": 5,
                      }
                    }
                  />
                  <br />
                  Checker 
                  <br />
                   tilgængelighed
                </span>
              </small>
            </div>
            <div
              className="title"
            >
              title1
            </div>
            <div
              className="creator"
            >
              creator1
            </div>
          </div>
        </div>
        <div
          className="row short-list-page"
        >
          <div
            className="col-12"
            style={
              Object {
                "borderTop": "1px solid #ccc",
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          >
            <span
              style={
                Object {
                  "float": "left",
                  "height": 60,
                  "marginRight": 10,
                }
              }
            >
              <BookCover
                book={
                  Object {
                    "creator": "creator2",
                    "pid": "pid2",
                    "title": "title2",
                  }
                }
                style={
                  Object {
                    "width": "unset",
                  }
                }
              />
            </span>
            <div
              style={
                Object {
                  "display": "inline-block",
                  "float": "right",
                  "height": 60,
                  "marginRight": 10,
                }
              }
            >
              <small>
                <span
                  style={
                    Object {
                      "display": "inline-block",
                      "textAlign": "center",
                    }
                  }
                >
                  <Spinner
                    size={12}
                    style={
                      Object {
                        "marginTop": 5,
                      }
                    }
                  />
                  <br />
                  Checker 
                  <br />
                   tilgængelighed
                </span>
              </small>
            </div>
            <div
              className="title"
            >
              title2
            </div>
            <div
              className="creator"
            >
              creator2
            </div>
          </div>
        </div>
        <div
          className="row short-list-page"
        >
          <div
            className="col-12"
            style={
              Object {
                "borderTop": "1px solid #ccc",
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          >
            <span
              style={
                Object {
                  "float": "left",
                  "height": 60,
                  "marginRight": 10,
                }
              }
            >
              <BookCover
                book={
                  Object {
                    "creator": "creator3",
                    "pid": "pid3",
                    "title": "title3",
                  }
                }
                style={
                  Object {
                    "width": "unset",
                  }
                }
              />
            </span>
            <div
              style={
                Object {
                  "display": "inline-block",
                  "float": "right",
                  "height": 60,
                  "marginRight": 10,
                }
              }
            >
              <small>
                <span
                  style={
                    Object {
                      "display": "inline-block",
                      "textAlign": "center",
                    }
                  }
                >
                  <Spinner
                    size={12}
                    style={
                      Object {
                        "marginTop": 5,
                      }
                    }
                  />
                  <br />
                  Checker 
                  <br />
                   tilgængelighed
                </span>
              </small>
            </div>
            <div
              className="title"
            >
              title3
            </div>
            <div
              className="creator"
            >
              creator3
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="form-group"
      style={
        Object {
          "marginBottom": 0,
        }
      }
    >
      <label
        htmlFor="pickupBranch"
      >
        Til afhentning på:
      </label>
      <select
        className="form-control"
        id="pickupBranch"
        onChange={undefined}
        style={
          Object {
            "width": "auto",
          }
        }
        value={undefined}
      />
    </div>
  </div>
</Modal>
`;
